### PR TITLE
Backports for 1.1.0-rc2

### LINKED ISSUE #29983: Uncommenting unused variable assignment (or other various things) gives different results
Using the code

```jl
using ForwardDiff: Dual
using StaticArrays

v = ((1.,2.), (2.,3.))

function f(x::Vector{T}) where {T}
    x1 = SVector((x[1],)) # comment this to get correct result
    la1 = SVector((x[1],))
    f1 = zero(SVector{1}) # can also initialize with zero(SVector{1, T}) to get correct result 
    for _ in v
        f1 += la1
        @show f1
    end
    return f1
end

x = [Dual(1.0,1.0)]
@show f(x)
```

gives the erroneous output:

```jl
f1 = Dual{Nothing,Float64,1}[Dual{Nothing}(1.0,1.0)]
f1 = Dual{Nothing,Float64,1}[Dual{Nothing}(1.0,1.0)]
f(x) = Dual{Nothing,Float64,1}[Dual{Nothing}(1.0,1.0)]
```

Now, we comment out the line `x1 = SVector((x[1],))`

```jl
f1 = Dual{Nothing,Float64,1}[Dual{Nothing}(1.0,1.0)]
f1 = Dual{Nothing,Float64,1}[Dual{Nothing}(2.0,2.0)]
f(x) = Dual{Nothing,Float64,1}[Dual{Nothing}(2.0,2.0)]
```

and get the correct result.

Executing the original code in the REPL also gives the correct result

```jl
julia> let
           x1 = SVector((x[1],))
           la1 = SVector((x[1],))
           f1 = zero(SVector{1})
           for _ in v
               f1 += la1
               @show f1
           end
           f1
       end
f1 = Dual{Nothing,Float64,1}[Dual{Nothing}(1.0,1.0)]
f1 = Dual{Nothing,Float64,1}[Dual{Nothing}(2.0,2.0)]
1-element SArray{Tuple{1},Dual{Nothing,Float64,1},1,1}:
 Dual{Nothing}(2.0,2.0)
```

If we initialize `f1` to be of type `T` we also get correct result.

My guess it is something with the type instability in combination with broadcasting in combination with optimization but not sure...

### PR DESCRIPTION
**Pull request by KristofferC**
Backported PRs:
- [x] #30483 - make inference of `_apply` match the implementation more closely
- [x] #30470 - use type inference world in `return_type`
- [x] #30599 - Use XCode 8.3 for macOS on Travis
- [x] #30598 - Fix use counts for mutable struct SROA
- [x] #30566 - some latency hacks
- [x] #30601 - Fix #29983
- [x] #30666 - Force specialization of the SubArray boundserror method.
- [x] #30497 - Add BB downloading for OpenBLAS
- [x] #30535 - Fix `bb-install` naming conventions, add hashes

Need manual backport:
- [ ] #30560 - Remove duplicate NEWS entry

Non-merged PRs with backport label:
- [ ] #30431 - Build: fix output directory when specified with O=
- [ ] #30403 - Include @​propagate_inbounds docstring in documentation
- [ ] #30398 - Fix enumerate documentation
- [ ] #30367 - move generate_precompile to Base so that it is bundled in downloads
- [ ] #30357 - workaround for F_DUPFD_CLOEXEC on Linux kernel <2.6.24 before it existed

### CODE REVIEW

**Comment by ararslan** [comment]
I've marked #30599 for backporting, which should fix the macOS CI. @staticfloat, it looks like we'll also need to backport whatever it is you did that fixed AppVeyor.

**Comment by staticfloat** [comment]
So probably https://github.com/JuliaLang/julia/commit/8b189ec494479d9e2ffda0549f066ba2c6389182 then

**Comment by KristofferC** [comment]
@nanosoldier `runbenchmarks("linalg", vs = ":release-1.1")`

**Comment by KristofferC** [comment]
@ararslan could you kick nanosoldier?

**Comment by staticfloat** [comment]
The appveyor failure is bizarre; `perl` is complaining that it can't make changes without a backup here: https://github.com/JuliaLang/julia/blob/346199d100ea106fa0f80a33ee4a3333a605a273/deps/blas.mk#L96

That line hasn't changed in 3 years.  Could `perl` have changed out from underneath us?

**Comment by staticfloat** [comment]
Also why are we even using `perl` here; we could just turn that into a normal `.patch`.

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/30607#issuecomment-452829815) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/d957a6949aac37a6de1e345f2967c7a329ccc6ce/dae984d_vs_ecf9911/report.md). cc @ararslan

**Comment by vtjnash** [comment]
> Also why are we even using perl here; we could just turn that into a normal .patch.

We could try, but this bug has been fixed since v0.1alpha2.5 over 7 years ago, so it would just be an empty file. (https://github.com/xianyi/OpenBLAS/commit/0d76196a099991b199b286c932b1a9919bcd0a48)

**Comment by KristofferC** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":release-1.1")`

**Comment by KristofferC** [comment]
I tried to just remove it.

File: deps/blas.mk [hunk 1]
```diff
@@ -92,10 +92,6 @@ endif
 # Do not overwrite the "-j" flag
 OPENBLAS_BUILD_OPTS += MAKE_NB_JOBS=0
 
-$(BUILDDIR)/$(OPENBLAS_SRC_DIR)/build-configured: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/source-extracted
-	perl -i -ple 's/^\s*(EXTRALIB\s*\+=\s*-lSystemStubs)\s*$$/# $$1/g' $(dir $<)/Makefile.system
```
Comment (staticfloat) [inline deps/blas.mk]: I think you still need to create the `build-configured` file though; so just remove the `perl` invocation

Comment (KristofferC) [reply deps/blas.mk]: Alright,  thanks.

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/30607#issuecomment-452899345) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/43ac38d25687f578eb62b74cc9ec22d540cc9a83/dae984d_vs_ecf9911/report.md). cc @ararslan

**Comment by KristofferC** [comment]
AV seems to build OpenBLAS from scratch now and times out during test on 32 bit but fails on 64 bit with:

```
cp: cannot stat 'scratch/openblas-fd8d1868a126bb9f12bbc43b36ee30d1ba943fbb/libopenblas64_.dll': No such file or directory
make[1]: *** [/cygdrive/c/projects/julia/deps/blas.mk:112: /cygdrive/c/projects/julia/usr-staging/openblas-fd8d1868a126bb9f12bbc43b36ee30d1ba943fbb.tgz] Error 1
make[1]: Leaving directory '/cygdrive/c/projects/julia/deps'
make: *** [Makefile:57: julia-deps] Error 2
```

Did I do something bad in the last commit? Help plz.

**Comment by ararslan** [comment]
PkgEval shows no regressions since RC1. Note that it was run prior to the most recent Makefile changes.

**Comment by staticfloat** [comment]
I'm not sure how, but it looks like you only got part of 87c18d88aefed9154af18aea9cdabca76763409e.  You need the rest of the changes as well.

**Comment by KristofferC** [comment]
Well, that wasn't marked with backport label and it wasn't the commit in  https://github.com/JuliaLang/julia/pull/30607#issuecomment-451718062 so it wasn't backported. I'll add that as well then. 

But why do we need so many build backports to this branch? Did something change on AV's end?

**Comment by KristofferC** [comment]
Backported those two but something is still wrong :(

```
make[1]: Leaving directory '/cygdrive/c/projects/julia'
make[1]: Entering directory '/cygdrive/c/projects/julia'
fatal: cannot change to '/cygdrive/c/projects/julia': No such file or directory
```

**Comment by staticfloat** [comment]
I have no idea what Appveyor is smoking there.  That doesn't make any sense to me, especially as we can see other processes happily going and entering that directory.

**Comment by KristofferC** [comment]
Ok, this happens on master as well. Doesn't keep the tests from running apparently.

**Pull request merged**